core/thread/credentials.go

```diff
@@ -0,0 +1,98 @@
+package thread
+
+import (
+	"github.com/libp2p/go-libp2p-core/crypto"
+)
+
+// Signature of an ID
+type Signature []byte
+
+// Credentials are used to prove a caller identity, which is needed
+// for thread ACL checks.
+type Credentials interface {
+	// ThreadID must return the signed thread ID.
+	ThreadID() ID
+
+	// Sign must return the public key and signature which can
+	// be used to verify the caller.
+	Sign() (crypto.PubKey, Signature, error)
```

Comment:
This method provides a way for a caller to conceal their private key for signing.

Reply:
Looking below, `Signature` is the `thread.ID` signed with the corresponding pk of `cyrpto.PubKey`.
This means that the signature is always the same for that thread.ID. This means that the same `Signature` can be reused as-is in different records.

I'm a bit confused at this point. Why is this necessary? Say, instead of doing the usual thing of the author signing whatever is sending and the caller to verify just that thing that was signed?
I can't imagine how using an "static" signature can be useful for other things.

I'm continuing on the review, so that might click later. Just wanted to share the thought by just looking at this interface!

Reply:
Yep, also forgot to mention this weirdness among all the other comments. I did start down the path of a more normal signature flow (the entire message envelope / body, as in the net layer), but then ran into an issue with restarting DBs. Since these signatures aren't needed yet (the ACL is not implemented), I just went with this simpler solution to get the API in. I think we'll want to have some _skeleton key_ or other way for a db to not need verification _under the hood_, in single thread adapter, but more to consider there.

Reply:
@carsonfarmer did my comment above answer your question about the static signature?